crates/flowctl/src/typescript/mod.rs

```diff
@@ -101,8 +100,75 @@ pub async fn do_generate(
         anyhow::bail!("errors while loading catalog sources");
     }
 
+    // It's possible that the sources may reference a collection that's meant to be resolved against live_specs.
+    // We'll collect a set of unresolved collection references, and then attempt to resolve them by fetching their
+    // specs from `live_specs`, and including them in the build.
+    let mut live_specs_catalog = models::Catalog::default();
+
+    for transform_row in source_tables.transforms.iter() {
+        let source_name = &transform_row.spec.source.name;
+        let is_included_in_sources = source_tables
+            .collections
+            .iter()
+            .find(|c| &c.collection == source_name)
+            .is_some();
+        let is_fetched = live_specs_catalog.collections.contains_key(source_name);
+        if !is_included_in_sources && !is_fetched {
+            let resolved = try_resolve_collection(ctx.controlplane_client()?, source_name).await.with_context(|| {
+                anyhow::anyhow!("resolving the collection '{source_name}', referenced from the transform '{}' in derivation '{}'", transform_row.transform, transform_row.derivation)
+            })?;
+            live_specs_catalog
+                .collections
+                .insert(source_name.clone(), resolved);
+        }
+    }
+
+    // If we've fetched any collection specs, then we'll need to load them now.
+    let resolved_tables = if !live_specs_catalog.collections.is_empty() {
+        // Write the bundled catalog to a temporary file. This is a little hacky, but Loader doesn't have a great way to
```

Comment:
Just for my own understanding when reading this comment, is there any actual file-writing going on here? It looks like we're serializing the catalog to bytes, and then loading those bytes from the "virtual" path `flowctl://resolved-catalog-sources`.

Reply:
Whoops, that was an old comment. I just updated it